### PR TITLE
fence all logs with debug flag

### PR DESCRIPTION
**Pull request by doapp-ryanp**
Only reason I can think of to log GA is when validating integration with the service, especially since we don't care about return codes.  Having it on by default adds alot of unnecessary noise to the log. fence all logs off with the debug flag to give consumers the choice to have em on or off.

### CODE REVIEW

File: roku-gamp/source/googleAnalytics.brs [hunk 1]
```diff
@@ -99,7 +101,9 @@ End Function
 '** Use the Event for application state events, such as a login or registration.
 '**
 Function gamobileEvent(category As String, action As String, label="" As String, value="" As String) As Void
-  ? "[GA] Event: " + category + "/" + action
+  if m.gamobile.enable <> true then
```
Comment (cdthompson) [inline roku-gamp/source/googleAnalytics.brs]: Were these supposed to be `gamobile.debug` ? 

Lines 89, 104, 121, 136

Comment (doapp-ryanp) [reply roku-gamp/source/googleAnalytics.brs]: Crap, sorry - fixed.

**Comment by cdthompson** [comment]
Two issues i found during testing

1. Line 162 is lacking a fence
2. Line 176 is also lacking a fence.

**Comment by doapp-ryanp** [comment]
162 was intentional because we don't want to suppress exceptions.

I'll add around 176

**Comment by cdthompson** [comment]
Please add 162 as well.  My argument is that this is not a logging subsystem which interprets severity.  Instead, this is simply reporting that GAMP is sending a report of that exception.  I'd much rather that we consistently log everything when debug is on, and nothing when debug is off.

**Comment by doapp-ryanp** [comment]
fair enough. done

**Comment by cdthompson** [comment]
Excellent.  Thanks for the contribution.

**Pull request merged**